core/src/main/scala/kafka/server/ApiVersionManager.scala

```diff
@@ -84,18 +105,30 @@ class SimpleApiVersionManager(
     throw new UnsupportedOperationException("This method is not supported in SimpleApiVersionManager, use apiVersionResponse(throttleTimeMs, finalizedFeatures, epoch) instead")
   }
 
-  override def apiVersionResponse(throttleTimeMs: Int, finalizedFeatures: Map[String, java.lang.Short], epoch: Long): ApiVersionsResponse = {
+  override def apiVersionResponse(throttleTimeMs: Int, finalizedFeatures: Map[String, java.lang.Short], finalizedFeaturesEpoch: Long): ApiVersionsResponse = {
     ApiVersionsResponse.createApiVersionsResponse(
       throttleTimeMs,
       apiVersions,
       brokerFeatures,
       finalizedFeatures.asJava,
-      epoch,
+      finalizedFeaturesEpoch,
       zkMigrationEnabled
     )
   }
 }
 
+/**
+ * The default ApiVersionManager that supports forwarding and has metadata cache, used in broker and zk controller.
+ * When forwarding is enabled, the enabled apis are determined by the broker listener type and the controller apis,
+ * otherwise the enabled apis are determined by the broker listener type, which is the same with SimpleApiVersionManager.
+ *
+ * @param listenerType the listener type
+ * @param forwardingManager the forwarding manager,
+ * @param features
+ * @param metadataCache
+ * @param enableUnstableLastVersion
+ * @param zkMigrationEnabled
```

Comment:
@dengziming thanks for adding the javadocs for these 2 classes. Very clear. But please remember to complete them. :)